makefiles/features_precheck.inc.mk

```diff
@@ -0,0 +1,15 @@
+# Check if provided features are available
+
+ifneq (,$(filter picolibc,$(FEATURES_PROVIDED)))
+  # Test if picolibc.specs is available
+  ifneq ($(shell $(LINK) -specs=picolibc.specs -E - 2>/dev/null >/dev/null </dev/null ; echo $$?),0)
+    FEATURES_BLACKLIST += picolibc
+ endif
+endif
```

Comment:
Why is there a need to check if the FEATURE is provided? Can't you simply blacklist if not present?

Reply:
as you guessed it is to avoid the shell call, if not need, and why should we blacklist something no one is talking about.